api/envoy/api/v2/route/route.proto

```diff
@@ -439,19 +439,24 @@ message RouteAction {
     google.protobuf.Duration per_try_timeout = 3 [(gogoproto.stdduration) = true];
   }
 
-  // Specifies the idle timeout for the route. If not specified, there is no
-  // stream level timeout. This is distinct to :ref:`timeout
+  // Specifies the idle timeout for the route. If not specified, this defaults
+  // to 5 minutes. The default value was select so as not to interfere with any
+  // smaller configured timeouts that may have existed in configurations prior
+  // to the introduction of this feature, while introducing robustness to TCP
+  // connections that terminate without FIN.
+  //
+  // The idle timeout is distinct to :ref:`timeout
   // <envoy_api_field_route.RouteAction.timeout>`, which provides an upper bound
   // on the upstream response time; :ref:`idle_timeout
   // <envoy_api_field_route.RouteAction.idle_timeout>` instead bounds the amount
   // of time the request's stream may be idle.
   //
-  // After header decoding, and assuming that this is not a WebSocket upgrade,
-  // the idle timeout will apply on downstream and upstream request events. Each
-  // time an encode/decode event for headers or data is processed for the
-  // stream, the timer will be reset. If the timeout fires, the stream is
-  // terminated with a 408 Request Timeout error code if no upstream response
-  // header has been received, otherwise a stream reset occurs.
+  // After header decoding, the idle timeout will apply on downstream and
```

Comment:
I know the existing hcm timeout applies when there are no streams.  Do we have an open window where we've started reading headers for stream but have not completed reading headers for a stream where we will still leak stale connections?

Because we can't do a route-specific timeout until we have complete headers I don't believe you could fix any timeout-gap in this PR, but if it'd still exist it'd be good to file a tracking issue.

Reply:
Agree; I think a global stream idle timeout makes sense to fix this. I've filed https://github.com/envoyproxy/envoy/issues/3853 and will do a followup PR once this one merges to introduce that.